lib/cc/analyzer/formatters/formatter.rb

```diff
@@ -2,6 +2,8 @@ module CC
   module Analyzer
     module Formatters
       class Formatter
+        attr_reader :current_engine
```

Comment:
Nope, that's the traditional definition of these keywords. In ruby `private` just means you can't call it with a receiver (`foo` works, `anything.foo`, `self.foo` errors). This means within a subclass you can call `current_engine` which is private and it works because there's no method receiver.

`protected` means you can only call it with a receiver that is your or your subclasses. It's meant for this scenario:

```rb
class Foo
  def ==(other)
    name == other.name
  end

  protected

  # this should be private, but because we need to invoke other.name 
  # within #==, we use protected
  attr_reader :name
end
```

> My main goal here is that this is accessible to subclasses but not outside.

Yup, done.